nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -307,6 +313,41 @@ private UpdateableReference loadReference(String requestedRef) {
     }
   }
 
+  private UpdateableReference loadReferenceAtTime(String requestedRef, Instant timestamp) {
+    try {
+      Reference ref = requestedRef == null ? client.getTreeApi().getDefaultBranch()
+          : client.getTreeApi().getReferenceByName(requestedRef);
+      if (ref instanceof Hash) {
+        LOGGER.warn("Cannot specify a hash {} and timestamp {} together. " +
+            "The timestamp is redundant and has been ignored", requestedRef, timestamp);
```

Comment:
I would rather fail when this happens than choose one argument to ignore.

Reply:
:+1:  fixed